loopy/codegen/instruction.py

```diff
@@ -29,24 +29,29 @@
 from loopy.codegen import Unvectorizable
 from loopy.codegen.result import CodeGenerationResult
 from pymbolic.mapper.stringifier import PREC_NONE
+from pytools import memoize_method
+
+
+@memoize_method
+def _get_new_implemented_domain(chk_domain, implemented_domain):
+
+    chk_domain, implemented_domain = isl.align_two(
+            chk_domain, implemented_domain)
+    chk_domain = chk_domain.gist(implemented_domain)
+
+    new_implemented_domain = implemented_domain & chk_domain
```

Comment:
> why would `chk_domain` and `new_implemented_domain` be any different?

The point is that the domain slicing code is allowed to overapproximate the loop domain (e.g. by only considering upper/lower bounds). So this here is the last layer of "filtering". If you have a stride-2 domain (say), here is where the `if` corresponding to "only use the even ones" gets inserted.